### ISSUE TITLE
suspicious_operation_groupings false positive with math code

### ISSUE THREAD

**Issue by IceSentry** [rust-lang/rust-clippy#7244]
Lint name:
suspicious_operation_groupings

I tried this code:

```rust
fn main() {
	// these are actually different computations, but the lint appears either way
	let a = 1.0;
	let b = 1.0;
	let c = 1.0;
	let discriminant = (b * b) - (a * c);
}
```

I expected to see this happen: Receive no warnings

Instead, this happened:
```
warning: this sequence of operators looks suspiciously like a bug
 --> src/main.rs:5:25
  |
5 |     let discriminant = (b * b) - (a * c);
  |                         ^^^^^ help: did you mean: `a * b`
  |
```
The suggestion isn't the math I need to do. If I use `b.powi(2)` it works, but I don't want to use a function call for a simple multiplication.

### Meta

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2ccfa7d1544271f1a2b4eaf52975a52b

**Comment by sereinity** [comment]
It looks like a duplicate of #6722 .

**Comment by IceSentry** [comment]
It does look similar, I'm not sure how I missed it when looking for it

**Comment by giraffate** [comment]
I'm closing this because of https://github.com/rust-lang/rust-clippy/issues/7244#issuecomment-849780222. Anyway, thanks for the report!
